### PR TITLE
feat(ui): Allows citizens to update their passport signer, also redesigned the passport view

### PR DESCRIPTION
**Pull request by luisivan**
Paves the way for citizens to be able to participate in votes on the go, or authenticate themselves at in-person events without carrying a private key with meaningful funds.

Screen Shot 2022-07-15 at 2 16 48 PM
Screen Shot 2022-07-15 at 2 21 59 PM

### CODE REVIEW

**Review by aahna-ashina — CHANGES_REQUESTED** [review]
Looks like the Vercel deployment failed.  I get a 404 error when trying to check why at https://vercel.com/nation3/app/6mvAybNNcWFT4p7L6YR3x7drHFm1

Screen Shot 2022-07-16 at 11 19 21 AM

**Comment by luisivan** [comment]
TypeScript stuff :D checking

**Review by aahna-ashina — CHANGES_REQUESTED** [review]
Screen Shot 2022-07-19 at 12 43 11 PM

**Comment by xPi2** [comment]
I personally don't like floating items over other floating items.
What about making this slight change?
Screenshot 2022-07-19 at 12 06 56

**Comment by aahna-ashina** [comment]
@luisivan For comparison:  Is this similar to vote delegation, where one account grants another account to vote on its behalf, or is a signer account a different type of thing?

**Review by gregfromstl — DISMISSED** [review]

**Review by xPi2 — APPROVED** [review]
Screenshot 2022-07-21 at 11 26 27
I'm having this curious bug on some page reloads, but maybe it's only on my end.

**Comment by luisivan** [comment]
@xPi2 I was having it too but I though it was a problem with my Web3 provider. Looks like it's not, will investigate

**Comment by gregfromstl** [comment]
@luisivan do you know when this PR will be finalized? I'd like to open a PR merging the migration to the `@nation3/utils` package but it conflicts with these changes.

**Comment by luisivan** [comment]
Now that I'm at it, I aligned all cards in the the mobile view to have the same width

Screen Shot 2022-07-22 at 8 59 58 AM

**Comment by xPi2** [comment]
Looks dope to me. I keep my approval.
Just review the conflicts before merging.

File: ui/lib/passport-nft.ts [hunk 1]
```diff
@@ -57,3 +58,32 @@ export function usePassport(address: any) {
   const { loading, nft } = useNft(nationPassportNFT || '', id)
   return { data: { id, nft }, isLoading: loadingID || loading }
 }
+
+export function usePassportSigner(id: number) {
+  console.log(id)
```
Comment (gregfromstl) [inline ui/lib/passport-nft.ts:63]: Can remove

**Review by gregfromstl — APPROVED** [review]

File: ui/lib/passport-nft.ts [hunk 2]
```diff
@@ -1,5 +1,6 @@
 import { useEffect, useState } from 'react'
 import { useNft } from 'use-nft'
+import PassporNFT from '../abis/Passport.json'
```
Comment (aahna-ashina) [inline ui/lib/passport-nft.ts:3]: Typo?
```suggestion
import PassportNFT from '../abis/Passport.json'
```

File: ui/lib/passport-nft.ts [hunk 3]
```diff
@@ -57,3 +58,32 @@ export function usePassport(address: any) {
   const { loading, nft } = useNft(nationPassportNFT || '', id)
   return { data: { id, nft }, isLoading: loadingID || loading }
 }
+
+export function usePassportSigner(id: number) {
+  console.log(id)
+  return useContractRead(
+    {
+      addressOrName: nationPassportNFT,
+      contractInterface: PassporNFT.abi,
```
Comment (aahna-ashina) [inline ui/lib/passport-nft.ts:67]:
```suggestion
      contractInterface: PassportNFT.abi,
```

File: ui/lib/passport-nft.ts [hunk 4]
```diff
@@ -57,3 +58,32 @@ export function usePassport(address: any) {
...
+      enable: id,
+    }
+  )
+}
+
+export function useSetPassportSigner(id: number, signerAddress: string) {
+  return useContractWrite(
+    {
+      addressOrName: nationPassportNFT,
+      contractInterface: PassporNFT.abi,
```
Comment (aahna-ashina) [inline ui/lib/passport-nft.ts:82]:
```suggestion
      contractInterface: PassportNFT.abi,
```

**Review by aahna-ashina — APPROVED** [review]

**Pull request merged**